Observer/ImportRates.php

```diff
@@ -358,41 +345,84 @@ private function makeOperation($rates, $bulkUuid): OperationInterface {
+     * @param string $bulkDescription
+     * @throws LocalizedException
+     */
+    private function schedule($bulkUuid, $operations, $bulkDescription): void
+    {
+        $userId = $this->userContext->getUserId();
+        $result = $this->bulkManagement->scheduleBulk($bulkUuid, $operations, $bulkDescription, $userId);
+
+        if (! $result) {
+            throw new LocalizedException(
+                __('Something went wrong while processing the request.')
+            );
+        }
+    }
+
     /**
      * Purge existing rule calculations and rates
      *
      * @return void
+     * @throws LocalizedException
      */
-    private function _purgeRates()
+    private function purgeRates()
     {
         $rateModel = $this->rateFactory->create();
         $rates = $rateModel->getExistingRates();
 
-        foreach ($rates as $rate) {
-            $calculations = $rateModel->getCalculationsByRateId($rate);
-
-            try {
-                foreach ($calculations->getItems() as $calculation) {
-                    // @codingStandardsIgnoreStart
-                    $calculation->delete();
-                    // @codingStandardsIgnoreEnd
-                }
-            } catch (\Exception $e) {
-                $this->messageManager->addErrorMessage($e->getMessage());
+        if ($rule = $rateModel->getRule()) {
+            $rule->getCalculationModel()->deleteByRuleId($rule->getId());
+            $rule->delete();
+        }
+
+        if (! empty($rates)) {
+            $chunkedRatesForDeletion = array_chunk($rates, $this->batchSize);
+            $bulkUuid = $this->identityService->generateId();
+
+            $operations = [];
+            foreach ($chunkedRatesForDeletion as $rateDeleteChunk) {
+                $operations[] = $this->makeOperation(
+                    $rateDeleteChunk,
+                    $bulkUuid,
+                    TaxjarConfig::TAXJAR_TOPIC_DELETE_RATES
+                );
             }
 
-            try {
-                $this->rateRepository->deleteById($rate);
-            } catch (\Exception $e) {
-                $this->messageManager->addErrorMessage($e->getMessage());
+            if (! empty($operations)) {
+                $bulkDescription = __('Delete %1 TaxJar backup tax rates.', $this->batchSize);
+                $this->schedule($bulkUuid, $operations, $bulkDescription);
             }
         }
     }
 
+    private function createRates(array $rates)
+    {
+        $rateChunks = array_chunk($rates, $this->batchSize);
+        $bulkUuid = $this->identityService->generateId();
+
+        $operations = [];
+        foreach ($rateChunks as $rateChunk) {
+            $operations[] = $this->makeOperation(
+                $rateChunk,
+                $bulkUuid,
+                TaxjarConfig::TAXJAR_TOPIC_CREATE_RATES
+            );
+        }
+
+        if (!empty($operations)) {
+            $bulkDescription = __('Create ' . $this->batchSize . ' TaxJar backup tax rates.');
```

Comment:
Similar to with the delete rates, the batch size is probably unnecessary in the description.